modules/site-health/audit-full-page-cache/load.php

```diff
@@ -0,0 +1,174 @@
+ *
+ * @package performance-lab
+ * @since 1.0.0
+ */
+
+/**
+ * Require helper functions.
+ */
+require_once __DIR__ . '/helper.php';
+
+/**
+ * Adds tests to site health.
+ *
+ * @since 1.0.0
+ *
+ * @param array $tests Site Health Tests.
+ * @return array
+ */
+function perflab_afpc_add_full_page_cache_test( $tests ) {
+	$tests['async']['perflab_page_cache'] = array(
+		'label'             => __( 'Page caching', 'performance-lab' ),
+		'test'              => rest_url( 'performance-lab/v1/tests/page-cache' ),
+		'has_rest'          => true,
+		'async_direct_test' => 'perflab_afpc_page_cache_test',
+	);
+	return $tests;
+}
+add_filter( 'site_status_tests', 'perflab_afpc_add_full_page_cache_test' );
+
+/**
+ * Callback for perflab_page_cache test.
+ *
+ * @since 1.0.0
+ *
+ * @return array
+ */
+function perflab_afpc_page_cache_test() {
+	$description  = '<p>' . __( 'Page caching enhances the speed and performance of your site by saving and serving static pages instead of calling for a page every time a user visits.', 'performance-lab' ) . '</p>';
+	$description .= '<p>' . __( 'Page caching is detected by looking for an active page caching plugin as well as making three requests to the homepage and looking for one or more of the following HTTP client caching response headers:', 'performance-lab' ) . '</p>';
+	$description .= '<code>' . implode( '</code>, <code>', array_keys( perflab_afpc_get_page_cache_headers() ) ) . '.</code>';
+
+	$result = array(
+		'badge'       => array(
+			'label' => __( 'Performance', 'performance-lab' ),
+			'color' => 'green',
+		),
+		'description' => wp_kses_post( $description ),
+		'test'        => 'perflab_page_cache',
+		'status'      => 'good',
+		'label'       => '',
+		'actions'     => sprintf(
+			'<p><a href="%1$s" target="_blank" rel="noopener noreferrer">%2$s <span class="screen-reader-text">%3$s</span><span aria-hidden="true" class="dashicons dashicons-external"></span></a></p>',
+			esc_url( 'https://wordpress.org/support/article/optimization/#Caching' ),
+			__( 'Learn more about page caching', 'performance-lab' ),
+			/* translators: The accessibility text. */
+			__( '(opens in a new tab)', 'performance-lab' )
+		),
+	);
+
+	$page_cache_detail = perflab_afpc_get_page_cache_detail();
+
+	if ( is_wp_error( $page_cache_detail ) ) {
+		$result['badge']['color'] = 'red';
+		$result['label']          = __( 'Unable to detect the presence of page caching', 'performance-lab' );
+		$result['status']         = 'critical';
+		$error_info               = sprintf(
+		/* translators: 1 is error message, 2 is error code */
+			__( 'Unable to detect page caching due to possible loopback request problem. Please verify that the loopback request test is passing. Error: %1$s (Code: %2$s)', 'performance-lab' ),
+			$page_cache_detail->get_error_message(),
+			$page_cache_detail->get_error_code()
+		);
+		$result['description'] = wp_kses_post( "<p>$error_info</p>" ) . $result['description'];
+		return $result;
+	}
```

Comment:
Should we even show the test when the loopback request is failing? It's a bit odd to not pass this test just because some other technical requirement for our own ability to conduct the test is missing.

My main point is: Having a page cache is crucial for performance. Loopback requests not working may not be that much of a problem depending on what you do with a site, at least it's not a performance problem. If loopback doesn't work, that means _we_ could not conduct this test - it doesn't mean that the site doesn't have page caching. So I think we should either not show the test in that situation at all or at least not mark it as critical and instead clarify that it is "N/A" because of the lack of loopback request support it couldn't be conducted.

Reply:
@felixarntz this was borrowed from the original code @westonruter can maybe elaborate further.

As far as I understand, if this error is triggered means the site is not accessible for whatever reason, since the loopback is just a wp_remote_get to the site.
This situation may not be realistic/reachable, or if that happens, the problem indeed may be somewhere else.

Reply:
Since there is another Site Health test for loopback requests, it would be redundant to also show an error in the page caching test for a loopback request failure. In that case, I think it would make sense for this to be downgraded from `critical` to either `recommended` with advice to look at the loopback request test to fix that issue since it is blocking this one, which the text of this is already doing.

Reply:
Thanks, @westonruter for your input, I agree. I've downgraded the warning to "recommended".